torch/csrc/dynamo/python_compiled_autograd.cpp

```diff
@@ -421,6 +422,20 @@ variable_list compiled_autograd(
 
       SwapSavedVariables saved(compiler_call, state);
       variable_list outputs = call.node->apply_with_saved(inputs, saved);
+      if (!call.post_acc_grad_hooks.empty()) {
```

Comment:
Don't do this here, in eager it is called in accumulate_grad.cpp.  We should do the same.

Reply:
For the purposes of show you the problem - It doesn't matter where call it, here or in accumulate_grad. It doesn't matter if we pass it through python with proxies, or if we directly call 

```
  auto& hook = tensor_post_acc_grad_hooks();
  if (hook != nullptr) {
    (*hook)(variable);
  }
```
from inside `apply_with_saved` - the .grad of the Tensor is `None` in either case in compiled_autograd, and I am trying to figure out why.

Reply:
Whats the right way to invoke the python fn here? 

Im doing this outside because I couldn't figure out the plumbing. I also tried to do it as an inductor op, but I dont think that supports hook signatures.

Reply:
Just follow the same pattern as the other hooks.

Reply:
I thought that's what I did here. post and pre hook are called above and below. No other hook is called from a AccumulateGrad or from within a node from what I saw?  I don't understand how to call a python hook from within the accumulate_grad implementation without plumbing in a bunch of python notions into it?  

Even if within `apply_with_saved` within AccumulateGrad we grab the hook and call it, you still need to pass a py_compiler from`the_autograd_compiler`, right? I didn't want to start mucking around with the `apply_with_saved` interface. Would you mind just giving me a little more detail on how you would do it here?

Reply:
Like, we could skip the py stuff and just do

```
auto& hook = tensor_post_acc_grad_hooks();
  if (hook != nullptr) {
    (*hook)(variable);
  }
```

But then it will bypass all the proxy and fake tensor stuff, which I don't think is what we want? Unless you want me to invoke the above with the fake tensor between before/after...?

Reply:
Well, I guess this works, but not sure if this is correct. I'll PR what I have and we can discuss it there.